commcare_export/cli.py

```diff
@@ -207,19 +218,99 @@ def _get_checkpoint_manager(args):
         return checkpoint_manager
 
 
+def evaluate_query(env, query):
+    with env:
+        try:
+            lazy_result = query.eval(env)
+            if lazy_result is not None:
+                # evaluate lazy results
+                for r in lazy_result:
+                    list(r) if r else r
+        except requests.exceptions.RequestException as e:
+            if e.response.status_code == 401:
+                print("\nAuthentication failed. Please check your credentials.")
+                return None
+            else:
+                raise
+        except KeyboardInterrupt:
+            print('\nExport aborted')
+            return None
+
+
+def location_type_map(args):
+    location_type_query = MiniLinq.from_jvalue(builtin_location_types_query.jvalue)
+
+    writer = writers.JValueTableWriter()
+    checkpoint_manager = None
+
+    commcarehq_base_url = commcare_hq_aliases.get(args.commcare_hq,
+                                                  args.commcare_hq)
+    api_client = _get_api_client(args, commcarehq_base_url)
+
+    cm = CheckpointManagerProvider(checkpoint_manager, None, args.start_over)
+    static_env = {
+        'commcarehq_base_url': commcarehq_base_url,
+        'get_checkpoint_manager': cm.get_checkpoint_manager,
+    }
+    env = (
+            BuiltInEnv(static_env)
+            | CommCareHqEnv(api_client, until=None, page_size=args.batch_size)
+            | JsonPathEnv({})
+            | EmitterEnv(writer)
+    )
+
+    evaluate_query(env, location_type_query)
+
+    return builtin_location_types_query.build_dictionary(writer.tables)
+
+
 def main_with_args(args):
     logger.info("CommCare Export Version {}".format(__version__))
     writer = _get_writer(args.output_format, args.output, args.strict_types)
 
-    try:
-        query = _get_query(args, writer)
-    except DataExportException as e:
-        print(e.message)
+    if args.query is None and args.users is False and args.locations is False:
+        print('At least one the following arguments is required: '
+              '--query, --users, --locations')
         return EXIT_STATUS_ERROR
 
-    if not query:
-        print('Query file not found: %s' % args.query)
-        return EXIT_STATUS_ERROR
+    query = None
+    query_tables = []
+    if args.query is not None:
+        try:
+            query = _get_query(args, writer)
+        except DataExportException as e:
+            print(e.message)
+            return EXIT_STATUS_ERROR
+
+        if not query:
+            print('Query file not found: %s' % args.query)
+            return EXIT_STATUS_ERROR
+        else:
+            query_tables = query.emitted_tables()
+
+    if args.users:
+        if builtin_users_query.TABLE_NAME in query_tables:
```

Comment:
rather than doing this and having to add all the `emitted_tables` function definitions I think we should just prevent exporting to this table via an excel query completely. This should be easy to do in this function by checking `sheet.name` against a blacklist: https://github.com/dimagi/commcare-export/blob/77d4a9e0ad61878bd9aaf35efa00a7922c63b415/commcare_export/excel_query.py#L451-L459

Reply:
Thanks for the suggestion. I made adding tables to the blacklist be a call from builtin_queries.py to avoid circular imports into excel_query.py.